protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,253 @@
+ *         Config is set before the next auction starts.
+ */
+contract SpiceAuction is ISpiceAuction, AuctionBase {
+    using SafeERC20 for IERC20;
+    using TempleMath for uint256;
+
+    /// @notice Spice auction contracts are set up for 2 tokens. Either can be bid or sell token for a given auction
+    /// @notice uint(TOKEN_A) < uint(TOKEN_B)
+    address public immutable override spiceToken;
+    /// @notice Temple GOLD
+    address public immutable override templeGold;
+    /// @notice DAO contract to execute configurations update
+    address public immutable override daoExecutor;
+
+    /// @notice Auctions run for minimum 1 week
+    uint32 public constant MINIMUM_AUCTION_PERIOD = 604_800;
+    /// @notice Name of this Spice Bazaar auction
+    string public override name;
+
+    /// @notice Last time auction was started. For zero auctions, it is the contract deploy timestamp
+    uint256 private immutable _deployTimestamp;
+
+    /// @notice Keep track of the different configurations for each auction
+    mapping(uint256 auctionId => SpiceAuctionConfig config) public auctionConfigs;
+    /// @notice Keep track of claimed amounts per auction token. This helps calculate amount of auction tokens for next epoch.
+    mapping(address token => uint256 amount) private _claimedAuctionTokens;
+    /// @notice Keep track of total allocation per auction token
+    mapping(address token => uint256 amount) private _totalAuctionTokenAllocation;
+
+    constructor(
+        address _templeGold,
+        address _spiceToken,
+        address _daoExecutor,
+        string memory _name
+    ) {
+        spiceToken = _spiceToken;
+        daoExecutor = _daoExecutor;
+        templeGold = _templeGold;
+        name = _name;
+        _deployTimestamp = block.timestamp;
+    }
+
+    /**
+     * @notice Set config for an epoch. This enables dynamic and multiple auctions especially for vested scenarios
+     * @dev Must be set before epoch auction starts
+     * @param _config Config to set
+     */
+    function setAuctionConfig(SpiceAuctionConfig calldata _config) external onlyDAOExecutor {
+        /// @dev epoch Id is only updated when auction starts. 
+        /// @dev cannot set config for past or ongoing auction
+        uint256 currentEpochIdCache = _currentEpochId;
+        if (currentEpochIdCache > 0) {
+            EpochInfo memory info = epochs[currentEpochIdCache];
+            /// Cannot set config for ongoing auction
+            if (info.startTime <= block.timestamp && block.timestamp < info.endTime) { revert InvalidConfigOperation(); }
+        }
+        if (_config.duration < MINIMUM_AUCTION_PERIOD) { revert CommonEventsAndErrors.InvalidParam(); }
+        // startCooldown can be zero
+        if (_config.waitPeriod == 0
+            || _config.minimumDistributedAuctionToken == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (_config.recipient == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+
```

Comment:
You could theoretically brick the auction from starting any new auctions for a long time if the `waitPeriod` is accidentally set to something stupidly high (eg 1 year). Perhaps worth adding sense checks to the caps on this and `duration`?